pictures/contrib/rest_framework.py

```diff
@@ -19,23 +19,45 @@ def default(obj):
 class PictureField(serializers.ReadOnlyField):
     """Read-only field for all aspect ratios and sizes of the image."""
 
+    def __init__(self, **kwargs):
+        self.aspect_ratio = kwargs.pop("aspect_ratio", None)
+        self.image_source = kwargs.pop("image_source", None)
+        super().__init__(**kwargs)
+
     def to_representation(self, obj: PictureFieldFile):
         if not obj:
             return None
-        payload = {
-            "url": obj.url,
-            "width": obj.width,
-            "height": obj.height,
-            "ratios": {
-                ratio: {
-                    "sources": {
-                        f"image/{file_type.lower()}": sizes
-                        for file_type, sizes in sources.items()
-                    },
-                }
-                for ratio, sources in obj.aspect_ratios.items()
-            },
-        }
+        if self.aspect_ratio and self.image_source:
```

Comment:
If you can do the same in a try, except, you might end up with improved performance. Conditional code blocks force the compiler to guess which instructions to load into the processor. Not a big issue on Apple Silicon, since the instruction cache is massive, but on other architectures, this can be an issue.

Reply:
This might send you down a very deep rabbit whole, without yielding substantial results. So, take this advice with caution.